client/src/app/pages/advisory-details/vulnerabilities.tsx

```diff
@@ -137,10 +130,14 @@ export const Vulnerabilities: React.FC<VulnerabilitiesProps> = ({
                       {item.non_normative.title}
                     </Td>
                     <Td width={10} {...getTdProps({ columnKey: "discovered" })}>
-                      {formatDate(item.non_normative.discovered)}
+                      {!!item.non_normative.discovered
+                        ? formatDate(item.non_normative.discovered)
+                        : null}
```

Comment:
why the unnecessary change of this block of code? what problem are we solving with this?

Reply:
@carlosthe19916, It's because I've made changes to https://github.com/trustification/trustify-ui/pull/70/files#diff-93d1e679f3a44569699332c49a4e0cf4e30ada958d50d897b3764cf2e20dd6f7R37-R39 where it makes more sense to not return `undefined` and therefore check for such case before calling `formatDate`.

Reply:
- Sorry but this change is replacing a simple line of code with a complex conditional longer code. This is is a step back in my opinion.
- What specific problem is this change solving that the previous code was not covering?

Reply:
@carlosthe19916, 

`formatDate(item.non_normative.discovered)` what happens when formatDate returns undefined ?
React might break in such case and what should be done is to trap such case and return null.

Reply:
- `formatDate` is already returning undefined in multiple cases. 
```shell
const myVar = formatDate(undefined); // Then myVar will be undefined

<>
{myVar}
</>
```

How is the previous code breaking reactjs? Can you give me a real concrete example and not suppositions?

-  I think the case is simple: Case1 vs Case2, let's explain why:

```shell
{!!item.non_normative.discovered
                        ? formatDate(item.non_normative.discovered)
                        : null}
```

is better than 

```
{formatDate(item.non_normative.released)}
```

Why if `item.non_normative.released` is undefined it will break the code?

- Couldn't we just do the following?:
```shell
export const formatDate = (value?: string) => {
  return value ? dayjs(value).format(RENDER_DATE_FORMAT) : null;
};
```

we are returning null as you want.

Reply:
@carlosthe19916 , sorry for the confusion, what I mean by breaking the code is not reactjs friendly when the code renders {undefined}.
Well it seems that React 18 handles undefined better. But that shouldn't really happen anyway : 
https://blog.saeloun.com/2022/04/14/react-18-allows-components-to-render-undfined/

Let's use your latest example and make sure formatDate and formatDateTime return null if undefined.